### PR TITLE
Db 429 cross file sql

### PR DESCRIPTION
**Pull request by bsweger**
Some updates to finish up DB-429, which implements cross file rules A18 and A19.

* Add SQL validation logic for cross-file validations.
* Update A18 and A19 rules in the SQL rules file (fix NULL handling)
* Add unittests for A18 and A19

### CODE REVIEW

File: dataactvalidator/validation_handlers/validator.py [hunk 1]
```diff
@@ -33,6 +33,36 @@ def crossValidate(cls,rules, submissionId):
...
+            if failedRows.rowcount:
+                # get list of fields involved in this validation
+                # note: row_number is metadata, not a field being
+                # validated, so exclude it
+                cols = failedRows.keys()
+                cols.remove('row_number')
+                columnString = ", ".join(str(c) for c in cols)
+                for row in failedRows:
+                    # get list of values for each column
+                    values = ["{}: {}".format(c, str(row[c])) for c in cols]
```
Comment (nmonga91) [inline dataactvalidator/validation_handlers/validator.py]: I'm assuming this is dynamically building a dictionary? Is that the correct understanding?

**Comment by nmonga91** [comment]
Looks good, merging!

File: dataactvalidator/validation_handlers/validator.py (hunk 1 above)
Comment (bsweger) [inline dataactvalidator/validation_handlers/validator.py]: It's dynamically building a list via [list comprehension](http://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/). The `{}` from the format statement make it look dictionary-like, though.

Comment (nmonga91) [inline dataactvalidator/validation_handlers/validator.py]: Gotcha, thanks!

**Comment by nmonga91** [comment]
Looks good, merging!

**Pull request merged**